### PR TITLE
Update Fabric introduction

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

- Shortens the introduction
- Adds a "Why Fabric" section in a reaction to feedback from users who were confused what Fabric was trying to solve that the Trainer not already did.

This requires review by @edenlightning 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: docs/source-pytorch/fabric/fabric.rst [hunk 1]
```diff
@@ -60,6 +54,33 @@ With only a few changes to your code, Fabric allows you to:
...
+**Maximum Flexibility**
+Wite your own training and/or inference logic down to the individual optimizer calls.
+You aren't forced to conform to a standardized epoch-based training loop like the one in :doc:`Lightning Trainer <../common/trainer>`.
+This also makes it super easy to adopt Fabric in existing PyTorch projects to speed-up and scale your models without the compromise on large refactors.
+Just remember: With great power comes a great responsibility.
+
+**Personalization**
+While a general-purpose trainer like :doc:`Lightning Trainer <../common/trainer>` contains all features *any* researcher could ever ask for,
+it may contain way more stuff than you, the individual, would ever need.
+This can make it more difficult to adapt it to your domain of research than it should be, but at the same time, building a well-tested, efficient and hackable trainer is very time-consuming.
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst]: Reformulated in [e3ba862](https://github.com/Lightning-AI/lightning/pull/16672/commits/e3ba862a1d23ea5fd6021b7d993d922c1d54aeed)

I removed the reference to "domain". It's more about trusting/debugging/studying the internals of someone else's Trainer vs. building your own that you know well.

File: docs/source-pytorch/fabric/fabric.rst [hunk 2]
```diff
@@ -60,6 +54,33 @@ With only a few changes to your code, Fabric allows you to:
...
 
+***********
+Why Fabric?
+***********
+
+Fabric differentiates itself from a fully-fledged trainer like :doc:`Lightning Trainer <../common/trainer>` in these key aspects:
+
+**Maximum Flexibility**
+Wite your own training and/or inference logic down to the individual optimizer calls.
+You aren't forced to conform to a standardized epoch-based training loop like the one in :doc:`Lightning Trainer <../common/trainer>`.
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst:66]: Added a sentece in Reformulated in [e3ba862](https://github.com/Lightning-AI/lightning/pull/16672/commits/e3ba862a1d23ea5fd6021b7d993d922c1d54aeed)

File: docs/source-pytorch/fabric/fabric.rst [hunk 3]
```diff
@@ -60,6 +55,34 @@ With only a few changes to your code, Fabric allows you to:
 ----
 
 
+***********
+Why Fabric?
+***********
+
+Fabric differentiates itself from a fully-fledged trainer like :doc:`Lightning Trainer <../common/trainer>` in these key aspects:
```
Comment (edenlightning) [inline docs/source-pytorch/fabric/fabric.rst:62]: The most important one we should add is Fast to implement- no need to restructure your code, just change a few lines in any pytorch native script and you'll be able to leverage Fabric features.

File: docs/source-pytorch/fabric/fabric.rst [hunk 4]
```diff
@@ -60,6 +55,34 @@ With only a few changes to your code, Fabric allows you to:
...
+**Personalization**
+While a general-purpose trainer like :doc:`Lightning Trainer <../common/trainer>` contains all features *any* researcher could ever ask for,
+it may contain way more stuff than you, the individual, would ever need.
+This makes it more difficult to study and trust the internals than it should be, but at the same time, building a well-tested, efficient and hackable trainer yourself is very time-consuming.
+Fabric bridges this gap by providing important tools to remove undesired boilerplate code (distributed, hardware, checkpoints, logging, ...), but leaves the design and orchestration fully up to you.
+
+**Opt-in Philosophy**
+Everything in Fabric is opt-in.
+Think of it as a toolbox: You take out the tools (Fabric functions) you need and leave the other ones behind.
+This makes it easier to develop and debug your PyTorch code as you gradually add more features to it.
```
Comment (edenlightning) [inline docs/source-pytorch/fabric/fabric.rst:80]: I would suggest combining the two
```suggestion
**Maximum Control**
The :doc:`Lightning Trainer <../common/trainer>` has many built in features to make research simpler with less boilerplate, but debugging it requires some familiarity with the framework internals.
In Fabric, everything is opt-in. Think of it as a toolbox: You take out the tools (Fabric functions) you need and leave the other ones behind.
This makes it easier to develop and debug your PyTorch code as you gradually add more features to it.
Fabric provided important tools to remove undesired boilerplate code (distributed, hardware, checkpoints, logging, ...), but leaves the design and orchestration fully up to you.
```

**Review by edenlightning — CHANGES_REQUESTED** [review]

File: docs/source-pytorch/fabric/fabric.rst (hunk 3 above)
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst:62]: added

File: docs/source-pytorch/fabric/fabric.rst (hunk 4 above)
Comment (awaelchli) [reply docs/source-pytorch/fabric/fabric.rst]: Added your text

File: docs/source-pytorch/fabric/fabric.rst [hunk 5]
```diff
@@ -2,19 +2,14 @@
...
-- Automatic support for mixed precision (speedup and smaller memory footprint)
-- Seamless switching between hardware (CPU, GPU, TPU)
-- State-of-the-art distributed training strategies (DDP, FSDP, DeepSpeed)
-- Easy-to-use launch command for spawning processes (DDP, torchelastic, etc)
-- Multi-node support (TorchElastic, SLURM, and more)
-- You keep complete control of your training loop
+- Handles all the boilerplate device logic for you
+- Easily switch from running on CPU to GPU (Apple Silicon, CUDA, ...), TPU, multi-GPU or even multi-node training
+- Brings useful tools to help you build a trainer (callbacks, logging, checkpoints, ...)
+- Designed with multi-billion parameter models in mind
```
Comment (edenlightning) [inline docs/source-pytorch/fabric/fabric.rst:10]:
```suggestion
- Easily switch from running on CPU to GPU (Apple Silicon, CUDA, ...), TPU, multi-GPU or even multi-node training
- State-of-the-art distributed training strategies (DDP, FSDP, DeepSpeed) and mixed precision out of the bix
- Handles all the boilerplate device logic for you
- Brings useful tools to help you build a trainer (callbacks, logging, checkpoints, ...)
- Designed with multi-billion parameter models in mind
```

**Review by edenlightning — APPROVED** [review]

**Pull request merged**